core/test/validation.jl

```diff
@@ -0,0 +1,23 @@
+using Ribasim
+using Graphs: DiGraph, add_edge!
+
+@testset "Neighbor count validation" begin
```

Comment:
While we're at it, we should probably also add a test for all (four) messages.

Since the errors are nicely concatenated, this can be done in a single test.